packages/ethereum-storage/src/ethereum-storage.ts

```diff
@@ -151,6 +151,23 @@ export class EthereumStorage implements StorageTypes.IStorage {
     }
   }
 
+  /**
+   * Function to create a NFT for one erc20 payment
+   *
+   * @param recipient the NFT owner
+   * @param assetToken erc20 token address
+   * @param tokenId NFT id, 64 bytes data
+   * @param metadata metadata string
+   */
+  public async tokenizeRequest(
+    recipient: string,
+    assetToken: string,
+    tokenId: string,
+    metadata: string,
+  ): Promise<any> {
+    return this.smartContractManager.mintInvoiceNFT(recipient, assetToken, tokenId, metadata);
```

Comment:
Yoann raised a great question in our meeting so I'm including it here for posterity: Why intertwine tokenization with ethereum storage? The answer is simply because ethereum-storage has an instantiated smartContractManager to call contracts with and we didn't want to instantiate it again somewhere else.